intellij/src/saros/intellij/ui/views/buttons/AddContactButton.java

```diff
@@ -0,0 +1,167 @@
+package saros.intellij.ui.views.buttons;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.InputValidator;
+import com.intellij.openapi.util.TextRange;
+import java.util.function.BiPredicate;
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import saros.SarosPluginContext;
+import saros.communication.connection.ConnectionHandler;
+import saros.communication.connection.IConnectionStateListener;
+import saros.exceptions.IllegalAWTContextException;
+import saros.exceptions.OperationCanceledException;
+import saros.intellij.ui.Messages;
+import saros.intellij.ui.util.IconManager;
+import saros.intellij.ui.util.SafeDialogUtils;
+import saros.net.ConnectionState;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.XMPPContactsService;
+import saros.repackaged.picocontainer.annotations.Inject;
+
+/** Button to add a contact to the currently connected account. */
+public class AddContactButton extends AbstractToolbarButton {
+  private static final Logger log = Logger.getLogger(AddContactButton.class);
+
+  @Inject private XMPPContactsService xmppContactsService;
+  @Inject private ConnectionHandler connectionHandler;
+
+  private final IConnectionStateListener connectionStateListener =
+      (state, error) -> {
+        if (state == ConnectionState.CONNECTED) {
+          setEnabled(true);
+        } else {
+          setEnabled(false);
+        }
+      };
+
+  public AddContactButton(@NotNull Project project) {
+    super(project, null, Messages.AddContactButton_tooltip, IconManager.ADD_CONTACT_ICON);
+
+    SarosPluginContext.initComponent(this);
+
+    connectionHandler.addConnectionStateListener(connectionStateListener);
+
+    addActionListener(actionEvent -> showAddContactDialog());
+
+    if (!connectionHandler.isConnected()) {
+      setEnabled(false);
+    }
+  }
+
+  @Override
+  public void dispose() {
+    connectionHandler.removeConnectionStateListener(connectionStateListener);
+  }
+
+  /**
+   * Shows the dialogs requesting the JID and (optional) nickname for the account to add as a
+   * contact.
+   *
+   * <p>Calls {@link XMPPContactsService#addContact(JID, String, BiPredicate)} if the dialogs were
+   * not canceled.
+   */
+  private void showAddContactDialog() {
+
+    // TODO better communicate issue to user
+    // TODO use checkInput as it disables the OK button instead of just doing nothing when OK is
+    //  clicked on illegal input.
+    InputValidator inputValidator =
+        new InputValidator() {
+          @Override
+          public boolean checkInput(String inputString) {
+            return true;
+          }
+
+          @Override
+          public boolean canClose(String inputString) {
+            JID jid = new JID(inputString);
+
+            return jid.isValid() && !jid.getName().isEmpty();
```

Comment:
Reading the IntelliJ Javadoc is not really helpful here. What is displayed when this method returns false ? Is there any hint ?

Reply:
Added javadoc stating what the method does. Also added an error dialog that is shown if the user presses "OK" while having entered an invalid JID. This gives feedback on why the dialog did not close.